WebService/Controllers/TaskController.cs

```diff
@@ -24,17 +25,28 @@ namespace WebService.Controllers
     public class TaskController : ControllerBase
     {
         private readonly ITaskContext _context;
+        private readonly IScheduler _scheduler;
 
-        public TaskController(ITaskContext context)
+        public TaskController(ITaskContext context, IScheduler scheduler)
         {
             _context = context;
+            _scheduler = scheduler;
         }
-        
+
         [HttpGet]
         [Route("ready")]
-        public Task GetReadyTask()
+        public IActionResult GetReadyTask([FromBody] ProviderDTO providerDto)
         {
-            return _context.FirstOrDefault(k => k.IsReady);
+            User user = providerDto.ToUser();

```

Comment:
Bare for det matcher en tidligere suggestion
```suggestion
            User user = providerDto.MapToUser();

```